aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -205,7 +205,16 @@ void div_mode_template(const Tensor& self, const Tensor& other,
     } else if (*rounding_mode == "trunc") {
       return trunc_tensor(mpsGraph, divTensor);
     } else if (*rounding_mode == "floor") {
-      return [mpsGraph floorWithTensor:divTensor name:nil];
+      MPSGraphTensor* floorTensor = [mpsGraph floorWithTensor:divTensor name:nil];
```

Comment:
```
moduloWithTensor
```
Denis can you check for Integer types if we just modulo it works fine ?
I checked on Numpy and it matches for integer types
```
 Number (-10, -9), mod -1, remainder -1
 Number (-10, -8), mod -2, remainder -2
 Number (-10, -7), mod -3, remainder -3
 Number (-10, -6), mod -4, remainder -4
 Number (-10, -5), mod 0, remainder 0
 Number (-10, -4), mod -2, remainder -2
 Number (-10, -3), mod -1, remainder -1
 Number (-10, -2), mod 0, remainder 0
 Number (-10, -1), mod 0, remainder 0
tmp.py:5: RuntimeWarning: divide by zero encountered in remainder
  mod, remainder = np.mod(*a), np.remainder(*a)
 Number (-10, 0), mod 0, remainder 0
 Number (-10, 1), mod 0, remainder 0
 Number (-10, 2), mod 0, remainder 0
 Number (-10, 3), mod 2, remainder 2
 Number (-10, 4), mod 2, remainder 2
 Number (-10, 5), mod 0, remainder 0
 Number (-10, 6), mod 2, remainder 2
 Number (-10, 7), mod 4, remainder 4
 Number (-10, 8), mod 6, remainder 6
 Number (-10, 9), mod 8, remainder 8
 Number (-9, -8), mod -1, remainder -1
 Number (-9, -7), mod -2, remainder -2
 Number (-9, -6), mod -3, remainder -3
 Number (-9, -5), mod -4, remainder -4
 Number (-9, -4), mod -1, remainder -1
 Number (-9, -3), mod 0, remainder 0
 Number (-9, -2), mod -1, remainder -1
 Number (-9, -1), mod 0, remainder 0
 Number (-9, 0), mod 0, remainder 0
 Number (-9, 1), mod 0, remainder 0
 Number (-9, 2), mod 1, remainder 1
 Number (-9, 3), mod 0, remainder 0
 Number (-9, 4), mod 3, remainder 3
 Number (-9, 5), mod 1, remainder 1
 Number (-9, 6), mod 3, remainder 3
 Number (-9, 7), mod 5, remainder 5
 Number (-9, 8), mod 7, remainder 7
 Number (-9, 9), mod 0, remainder 0
 Number (-8, -7), mod -1, remainder -1
 Number (-8, -6), mod -2, remainder -2
 Number (-8, -5), mod -3, remainder -3
 Number (-8, -4), mod 0, remainder 0
 Number (-8, -3), mod -2, remainder -2
 Number (-8, -2), mod 0, remainder 0
 Number (-8, -1), mod 0, remainder 0
 Number (-8, 0), mod 0, remainder 0
 Number (-8, 1), mod 0, remainder 0
 Number (-8, 2), mod 0, remainder 0
 Number (-8, 3), mod 1, remainder 1
 Number (-8, 4), mod 0, remainder 0
 Number (-8, 5), mod 2, remainder 2
 Number (-8, 6), mod 4, remainder 4
 Number (-8, 7), mod 6, remainder 6
 Number (-8, 8), mod 0, remainder 0
 Number (-8, 9), mod 1, remainder 1
 Number (-7, -6), mod -1, remainder -1
 Number (-7, -5), mod -2, remainder -2
 Number (-7, -4), mod -3, remainder -3
 Number (-7, -3), mod -1, remainder -1
 Number (-7, -2), mod -1, remainder -1
 Number (-7, -1), mod 0, remainder 0
 Number (-7, 0), mod 0, remainder 0
 Number (-7, 1), mod 0, remainder 0
 Number (-7, 2), mod 1, remainder 1
 Number (-7, 3), mod 2, remainder 2
 Number (-7, 4), mod 1, remainder 1
 Number (-7, 5), mod 3, remainder 3
 Number (-7, 6), mod 5, remainder 5
 Number (-7, 7), mod 0, remainder 0
 Number (-7, 8), mod 1, remainder 1
 Number (-7, 9), mod 2, remainder 2
 Number (-6, -5), mod -1, remainder -1
 Number (-6, -4), mod -2, remainder -2
 Number (-6, -3), mod 0, remainder 0
 Number (-6, -2), mod 0, remainder 0
 Number (-6, -1), mod 0, remainder 0
 Number (-6, 0), mod 0, remainder 0
 Number (-6, 1), mod 0, remainder 0
 Number (-6, 2), mod 0, remainder 0
 Number (-6, 3), mod 0, remainder 0
 Number (-6, 4), mod 2, remainder 2
 Number (-6, 5), mod 4, remainder 4
 Number (-6, 6), mod 0, remainder 0
 Number (-6, 7), mod 1, remainder 1
 Number (-6, 8), mod 2, remainder 2
 Number (-6, 9), mod 3, remainder 3
 Number (-5, -4), mod -1, remainder -1
 Number (-5, -3), mod -2, remainder -2
 Number (-5, -2), mod -1, remainder -1
 Number (-5, -1), mod 0, remainder 0
 Number (-5, 0), mod 0, remainder 0
 Number (-5, 1), mod 0, remainder 0
 Number (-5, 2), mod 1, remainder 1
 Number (-5, 3), mod 1, remainder 1
 Number (-5, 4), mod 3, remainder 3
 Number (-5, 5), mod 0, remainder 0
 Number (-5, 6), mod 1, remainder 1
 Number (-5, 7), mod 2, remainder 2
 Number (-5, 8), mod 3, remainder 3
 Number (-5, 9), mod 4, remainder 4
 Number (-4, -3), mod -1, remainder -1
 Number (-4, -2), mod 0, remainder 0
 Number (-4, -1), mod 0, remainder 0
 Number (-4, 0), mod 0, remainder 0
 Number (-4, 1), mod 0, remainder 0
 Number (-4, 2), mod 0, remainder 0
 Number (-4, 3), mod 2, remainder 2
 Number (-4, 4), mod 0, remainder 0
 Number (-4, 5), mod 1, remainder 1
 Number (-4, 6), mod 2, remainder 2
 Number (-4, 7), mod 3, remainder 3
 Number (-4, 8), mod 4, remainder 4
 Number (-4, 9), mod 5, remainder 5
 Number (-3, -2), mod -1, remainder -1
 Number (-3, -1), mod 0, remainder 0
 Number (-3, 0), mod 0, remainder 0
 Number (-3, 1), mod 0, remainder 0
 Number (-3, 2), mod 1, remainder 1
 Number (-3, 3), mod 0, remainder 0
 Number (-3, 4), mod 1, remainder 1
 Number (-3, 5), mod 2, remainder 2
 Number (-3, 6), mod 3, remainder 3
 Number (-3, 7), mod 4, remainder 4
 Number (-3, 8), mod 5, remainder 5
 Number (-3, 9), mod 6, remainder 6
 Number (-2, -1), mod 0, remainder 0
 Number (-2, 0), mod 0, remainder 0
 Number (-2, 1), mod 0, remainder 0
 Number (-2, 2), mod 0, remainder 0
 Number (-2, 3), mod 1, remainder 1
 Number (-2, 4), mod 2, remainder 2
 Number (-2, 5), mod 3, remainder 3
 Number (-2, 6), mod 4, remainder 4
 Number (-2, 7), mod 5, remainder 5
 Number (-2, 8), mod 6, remainder 6
 Number (-2, 9), mod 7, remainder 7
 Number (-1, 0), mod 0, remainder 0
 Number (-1, 1), mod 0, remainder 0
 Number (-1, 2), mod 1, remainder 1
 Number (-1, 3), mod 2, remainder 2
 Number (-1, 4), mod 3, remainder 3
 Number (-1, 5), mod 4, remainder 4
 Number (-1, 6), mod 5, remainder 5
 Number (-1, 7), mod 6, remainder 6
 Number (-1, 8), mod 7, remainder 7
 Number (-1, 9), mod 8, remainder 8
 Number (0, 1), mod 0, remainder 0
 Number (0, 2), mod 0, remainder 0
 Number (0, 3), mod 0, remainder 0
 Number (0, 4), mod 0, remainder 0
 Number (0, 5), mod 0, remainder 0
 Number (0, 6), mod 0, remainder 0
 Number (0, 7), mod 0, remainder 0
 Number (0, 8), mod 0, remainder 0
 Number (0, 9), mod 0, remainder 0
 Number (1, 2), mod 1, remainder 1
 Number (1, 3), mod 1, remainder 1
 Number (1, 4), mod 1, remainder 1
 Number (1, 5), mod 1, remainder 1
 Number (1, 6), mod 1, remainder 1
 Number (1, 7), mod 1, remainder 1
 Number (1, 8), mod 1, remainder 1
 Number (1, 9), mod 1, remainder 1
```

Reply:
Make sure you have to remove the cast Operation at the beginning of the function.

Reply:
> ```
> moduloWithTensor
> ```
> 
> Denis can you check for Integer types if we just modulo it works fine ? I checked on Numpy and it matches for integer types
> 
> ```
>  Number (-10, -9), mod -1, remainder -1

Python's/Numpy's modulo operator is not the same as the C modulo operator. 
In Python, `-8 % 6` will yield **4**, while in C `-8 % 6` will yield **-2**. `moduloWithTensor` follows C's standard and it will produce different results compared to Python's/Numpy's `mod` functions.